### PR TITLE
Move Cart, Checkout and Order Confirmation template customization tests to the parameterized tests file

### PR DESCRIPTION
**Pull request by Aljullu**
### Changes proposed in this Pull Request:

In https://github.com/woocommerce/woocommerce/pull/43778, we converted block templates tests to parameterized tests, but it didn't include Cart, Checkout and Order Confirmation templates tests.

### How to test the changes in this Pull Request:

Note: no need to test this for the release.

1. Verify Playwright e2e tests pass.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment

Move Cart, Checkout and Order Confirmation template customization tests to the parameterized tests file

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/tests/e2e/tests/templates/template-user-customization.block_theme.spec.ts [hunk 1]
```diff
@@ -47,11 +73,41 @@ const templateUserCustomizationTests = [
...
+			await frontendUtils.addToCart();
+			await frontendUtils.goToCheckout();
+		},
+		templateName: 'Page: Checkout',
+		templatePath: 'woocommerce/woocommerce//page-checkout',
+		templateType: 'wp_template',
+	},
+	{
+		visitPage: async ( { frontendUtils, page } ) => {
+			const checkoutPage = new CheckoutPage( { page } );
```
Comment (Aljullu) [inline plugins/woocommerce-blocks/tests/e2e/tests/templates/template-user-customization.block_theme.spec.ts:100]: I don't know if this is the best way to create the `CheckoutPage()` object, but it was the cleanest way I could find without having to extend the `test` object of all tests. :thinking:

File: plugins/woocommerce-blocks/tests/e2e/tests/templates/order-confirmation.block_theme.spec.ts [hunk 2]
```diff
@@ -85,37 +85,4 @@ test.describe( 'Test the order confirmation template', async () => {
 				} )
 		).toBeVisible();
 	} );
-
-	test( 'Template can be modified', async ( {
```
Comment (roykho) [inline plugins/woocommerce-blocks/tests/e2e/tests/templates/order-confirmation.block_theme.spec.ts:89]: In removing this code, perhaps we can clean up removing line 10-14 and 25-27 as they're no longer used?

**Review by roykho — CHANGES_REQUESTED** [review]

File: plugins/woocommerce-blocks/tests/e2e/tests/templates/order-confirmation.block_theme.spec.ts (hunk 2 above)
Comment (Aljullu) [reply plugins/woocommerce-blocks/tests/e2e/tests/templates/order-confirmation.block_theme.spec.ts:89]: Oh, good point, thanks! That file could be definitely cleaned up. Done in: 3eafde5f1dcabe601e2a5c1333d164d94adae38e.

**Review by roykho — APPROVED** [review]
LGTM!

**Pull request merged**